File: LCX/ValidEndPoints/dsl/examples/authenticated_trading/authenticated_trading.dart
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'dart:async';


Future<Map<String, dynamic>> get_account_info() async {
  // GET /api/account
  final url = Uri.parse('https://exchange-api.lcx.com/api/account');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/account';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_balances() async {
  // GET /api/balances
  final url = Uri.parse('https://exchange-api.lcx.com/api/balances');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/balances';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_asset_balance(asset) async {
  // GET /api/balance
  final url = Uri.parse('https://exchange-api.lcx.com/api/balance');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/balance';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_trading_fees() async {
  // GET /api/fees
  final url = Uri.parse('https://exchange-api.lcx.com/api/fees');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/fees';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_open_orders(offset, pair) async {
  // GET /api/open
  final url = Uri.parse('https://exchange-api.lcx.com/api/open');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/open';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_order_details(orderId) async {
  // GET /api/order
  final url = Uri.parse('https://exchange-api.lcx.com/api/order');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/order';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_order_history(offset, pair, fromDate, toDate, side, orderStatus, orderType) async {
  // GET /api/orderHistory
  final url = Uri.parse('https://exchange-api.lcx.com/api/orderHistory');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/orderHistory';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_user_trade_history(offset, pair, fromDate, toDate) async {
  // GET /api/uHistory
  final url = Uri.parse('https://exchange-api.lcx.com/api/uHistory');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/uHistory';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {
      return jsonDecode(response.body);

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# POST method not supported for dart
# PUT method not supported for dart
# DELETE method not supported for dart
# DELETE method not supported for dart
Future<Map<String, dynamic>> check_balance_workflow() async {
  // GET /api/balances
  final url = Uri.parse('https://exchange-api.lcx.com/api/balances');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/balances';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for dart
Future<Map<String, dynamic>> place_order_workflow(BTC/USDC) async {
  // GET /api/ticker
  final url = Uri.parse('https://exchange-api.lcx.com/api/ticker');
  final headers = {'Content-Type': 'application/json'};

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for dart
Future<Map<String, dynamic>> get_order_info(order_response) async {
  // GET /api/order
  final url = Uri.parse('https://exchange-api.lcx.com/api/order');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/order';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# None method not supported for dart
Future<Map<String, dynamic>> manage_open_orders_workflow(0) async {
  // GET /api/orderHistory
  final url = Uri.parse('https://exchange-api.lcx.com/api/orderHistory');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/orderHistory';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# Workflow: ManageOpenOrdersWorkflow (sequence)

Future<Map<String, dynamic>> get_trade_history(0) async {
  // GET /api/uHistory
  final url = Uri.parse('https://exchange-api.lcx.com/api/uHistory');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/uHistory';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# None method not supported for dart
# PUT method not supported for dart
# DELETE method not supported for dart
# None method not supported for dart
Future<Map<String, dynamic>> account_monitoring_workflow() async {
  // GET /api/balances
  final url = Uri.parse('https://exchange-api.lcx.com/api/balances');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/balances';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# Workflow: AccountMonitoringWorkflow (sequence)

Future<Map<String, dynamic>> get_open_orders(0) async {
  // GET /api/open
  final url = Uri.parse('https://exchange-api.lcx.com/api/open');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/open';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

Future<Map<String, dynamic>> get_fees() async {
  // GET /api/fees
  final url = Uri.parse('https://exchange-api.lcx.com/api/fees');
  final headers = {'Content-Type': 'application/json'};
  final apiKey = String.fromEnvironment('LCX_API_KEY');
  final apiSecret = String.fromEnvironment('LCX_API_SECRET');
  final timestamp = DateTime.now().millisecondsSinceEpoch.toString();
  final message = timestamp + 'GET' + '/api/fees';

  try {
    final response = await http.get(url, headers: headers);
    if (response.statusCode == 200) {

    } else {
      throw Exception('Failed to load data: ${response.statusCode}');
    }
  } catch (e) {
    print('Error: $e');
    rethrow;
  }
}

# None method not supported for dart

File: LCX/ValidEndPoints/dsl/examples/authenticated_trading/authenticated_trading.php
<?php
error_reporting(E_ALL);
ini_set('display_errors', 1);


function get_account_info() {
    // GET /api/account
    $url = "https://exchange-api.lcx.com/api/account";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/account";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_balances() {
    // GET /api/balances
    $url = "https://exchange-api.lcx.com/api/balances";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/balances";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_asset_balance(asset) {
    // GET /api/balance
    $url = "https://exchange-api.lcx.com/api/balance";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/balance";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_trading_fees() {
    // GET /api/fees
    $url = "https://exchange-api.lcx.com/api/fees";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/fees";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_open_orders(offset, pair) {
    // GET /api/open
    $url = "https://exchange-api.lcx.com/api/open";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/open";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_order_details(orderId) {
    // GET /api/order
    $url = "https://exchange-api.lcx.com/api/order";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/order";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_order_history(offset, pair, fromDate, toDate, side, orderStatus, orderType) {
    // GET /api/orderHistory
    $url = "https://exchange-api.lcx.com/api/orderHistory";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/orderHistory";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

function get_user_trade_history(offset, pair, fromDate, toDate) {
    // GET /api/uHistory
    $url = "https://exchange-api.lcx.com/api/uHistory";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/uHistory";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }
    return json_decode($response, true);

}

# POST method not supported for php
# PUT method not supported for php
# DELETE method not supported for php
# DELETE method not supported for php
function check_balance_workflow() {
    // GET /api/balances
    $url = "https://exchange-api.lcx.com/api/balances";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/balances";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for php
function place_order_workflow(BTC/USDC) {
    // GET /api/ticker
    $url = "https://exchange-api.lcx.com/api/ticker";
    $headers = ["Content-Type: application/json"];

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for php
function get_order_info(order_response) {
    // GET /api/order
    $url = "https://exchange-api.lcx.com/api/order";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/order";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# None method not supported for php
function manage_open_orders_workflow(0) {
    // GET /api/orderHistory
    $url = "https://exchange-api.lcx.com/api/orderHistory";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/orderHistory";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# Workflow: ManageOpenOrdersWorkflow (sequence)

function get_trade_history(0) {
    // GET /api/uHistory
    $url = "https://exchange-api.lcx.com/api/uHistory";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/uHistory";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# None method not supported for php
# PUT method not supported for php
# DELETE method not supported for php
# None method not supported for php
function account_monitoring_workflow() {
    // GET /api/balances
    $url = "https://exchange-api.lcx.com/api/balances";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/balances";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# Workflow: AccountMonitoringWorkflow (sequence)

function get_open_orders(0) {
    // GET /api/open
    $url = "https://exchange-api.lcx.com/api/open";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/open";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

function get_fees() {
    // GET /api/fees
    $url = "https://exchange-api.lcx.com/api/fees";
    $headers = ["Content-Type: application/json"];
    $apiKey = getenv("LCX_API_KEY");
    $apiSecret = getenv("LCX_API_SECRET");
    $timestamp = (string)round(microtime(true) * 1000);
    $message = $timestamp . "GET" . "/api/fees";
    $signature = base64_encode(hash_hmac('sha256', $message, $apiSecret, true));

    array_push($headers, "x-access-key: $apiKey");
    array_push($headers, "x-access-sign: $signature");
    array_push($headers, "x-access-timestamp: $timestamp");

    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($httpCode !== 200) {
        throw new Exception("HTTP Error: $httpCode");
    }

}

# None method not supported for php
